src/workerd/io/worker-entrypoint.c++

```diff
@@ -3,15 +3,108 @@
 //     https://opensource.org/licenses/Apache-2.0
 
 #include "worker-entrypoint.h"
+#include "io-context.h"
 #include <capnp/message.h>
 #include <workerd/jsg/jsg.h>
-#include <workerd/util/sentry.h>
 #include <workerd/api/global-scope.h>
 #include <workerd/api/util.h>
+#include <workerd/util/sentry.h>
 #include <workerd/util/thread-scopes.h>
+#include <kj/compat/http.h>
 
 namespace workerd {
 
+namespace {
+// Wrapper around a Worker that handles receiving a new event from the outside. In particular,
+// this handles:
+// - Creating a IoContext and making it current.
```

Comment:
This comment (roughly L18 - 24) is now duplicated and used in both files. You can keep it as-is if you think it's useful for describing the code, but something like only having it in the header would also be an option.